### ISSUE TITLE
2022/01/13 - 2022/01/20

### ISSUE THREAD

**Issue by danbi5228** [njs03332/nlp_study#16]
* 1월 20일 pm 9:00
* 다음 진도
   * 유리님: 5.2
   * 선미님: 5.3
   * 단비님: 4.2 끝까지! (씁)

**Comment by njs03332** [comment]
### 4.4 예제: CNN으로 성씨 분류하기
#### 4.4.2 Vocabulary, Vectorizer, DataLoader
- `Conv1d` 층에 넣는 데이터 텐서의 형태는 아래와 같아야 함
  - 0 차원: batch
  - 1 차원: channel
  - 2 차원: feature
  - SurnameVectorizer에서
    - `one_hot_matrix_size = (len(self.surname_vocab), self._max_surname_length)`
    - 행 크기: 가장 긴 성씨 길이
    - 열 크기: 원핫벡터 크기 (vocabulary 크기)
    - 여기에 첫번째 차원에 batch가 추가되면 위의 차원 조건을 만족
  - 136쪽 코드 4-14에서도
    - `data = torch.randn(batch_size, one_hot_size, sequence_width)`

**Comment by givitallugot** [comment]
#### ReduceLROnPlateau
- 더이상 학습이 진행되지 않을 때 learning rate를 감소시키는 scheduler
```python
scheduler = optim.lr_scheduler.ReduceLROnPlateau(optimizer=optimizer,
                                                 mode='min', factor=0.5,
                                                 patience=1)
```
- mode: metric이 낮을수록 좋은지('min') 높을수록 좋은지('max')
- patience: metric이 얼마 동안 변화가 없을 때 learning rate를 감소시킬지 결정

### 5.3 NewsClassifier
- glove 임베딩 사용하여 뉴스 기사 제목을 바탕으로 클래스(스포츠, 과학/기술, 세계, 비즈니스) 분류
